lib/ansible/modules/hostname.py

```diff
@@ -553,61 +460,69 @@ def set_permanent_hostname(self, name):
             self.module.fail_json(msg="Command failed rc=%d, out=%s, err=%s" % (rc, out, err))
 
 
-class FreeBSDStrategy(GenericStrategy):
+class FreeBSDStrategy(BaseStrategy):
     """
     This is a FreeBSD hostname manipulation strategy class - it edits
     the /etc/rc.conf.d/hostname file.
     """
 
-    HOSTNAME_FILE = '/etc/rc.conf.d/hostname'
+    FILE = '/etc/rc.conf.d/hostname'
+    COMMAND = "hostname"
+
+    def __init__(self, module):
+        super(FreeBSDStrategy, self).__init__(module)
+        self.hostname_cmd = self.module.get_bin_path(self.COMMAND, True)
+
+    def get_current_hostname(self):
+        cmd = [self.hostname_cmd]
+        rc, out, err = self.module.run_command(cmd)
+        if rc != 0:
+            self.module.fail_json(msg="Command failed rc=%d, out=%s, err=%s" % (rc, out, err))
+        return to_native(out).strip()
+
+    def set_current_hostname(self, name):
+        cmd = [self.hostname_cmd, name]
+        rc, out, err = self.module.run_command(cmd)
+        if rc != 0:
+            self.module.fail_json(msg="Command failed rc=%d, out=%s, err=%s" % (rc, out, err))
 
     def get_permanent_hostname(self):
+        if not os.path.isfile(self.FILE):
+            return ''
 
-        name = 'UNKNOWN'
-        if not os.path.isfile(self.HOSTNAME_FILE):
-            try:
-                open(self.HOSTNAME_FILE, "a").write("hostname=temporarystub\n")
-            except IOError as e:
-                self.module.fail_json(msg="failed to write file: %s" %
-                                          to_native(e), exception=traceback.format_exc())
         try:
-            try:
-                f = open(self.HOSTNAME_FILE, 'r')
+            with open(self.FILE, 'r') as f:
```

Comment:
Just a thought - can we use [`get_file_lines`](https://github.com/ansible/ansible/blob/015331518dff60f31a7d8ce24fc315e3ac9e86f8/lib/ansible/module_utils/facts/utils.py#L63) instead?

Reply:
sure, done